crates/ra_hir/src/code_model_impl/module.rs

```diff
@@ -133,7 +133,7 @@ impl Module {
             };
             let module = match curr.resolve(db)? {
                 Def::Module(it) => it,
-                // TODO here would be the place to handle enum variants...
+                Def::EnumVariant(ev) => return Ok(PerNs::values(ev.def_id)),
```

Comment:
I don't think that's the right thing -- the case you need to handle here is if `curr` resolves to an enum. E.g. if we have the path `Option::None`, we'll resolve `Option` to the enum here, and then we need to resolve the next segment as a variant of that enum. Once we are at the variant, we're fine (since that should be the last part of the path).

Reply:
That makes sense - will fix. Thanks!